src/backends/aws_s3.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @module Backends
+ */
+import { FetchFunction } from '../fetch';
+import aws, { Credentials } from '../aws'
+
+/**
+ * AWS S3 bucket options
+ */
+export interface AwsS3Options {
+    bucket: string,
+    region?: string,
+    credentials?: Credentials
+}
+
+/**
+ * Creates a fetch-like proxy function for making requests to AWS S3.
+ * 
+ * Example:
+ * 
+ * ```typescript
+ * import { awsS3 } from "./src/backends";
+ * const backend = awsS3({
+ *  bucket: "flyio-test-website",
+ *  // region: "us-east-1"
+ *  // credentials: { // for private S3 buckets
+ *  //   accessKeyId: app.config.aws_access_key_id,
+ *  //   secretAccessKey: app.config.aws_secret_access_key, // store this as a secret
+ *  // }
+ * });
+ * ```
+ * @param options AWS S3 bucket to proxy to
+ * @module Backends
+ */
+export function awsS3(options: AwsS3Options | string): FetchFunction {
+    const opts = normalizeOptions(options);
+    return async function awsS3Fetch(req: RequestInfo, init?: RequestInit): Promise<Response> {
+        if (typeof req === "string") req = new Request(req, init);
+        const url = new URL(req.url);
```

Comment:
When people request `/` we should infer `index.html`, at least on private buckets. This could be configurable pretty easily but I don't think we should ever get `path/` since S3 will happily just list files there.